theseus/core/robust_cost_function.py

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List, Tuple
+
+import torch
+
+import theseus as th
+
+from .loss import Loss
+
+
+class RobustCostFunction(th.CostFunction):
```

Comment:
Add unit testing for this.

Reply:
Will do. I'm not considering to merge this branch to main. This is only for profiling and benchmarking.